### ISSUE TITLE
Optional fields are added inside a group

### ISSUE THREAD

**Issue by SamuelYvon** [danielgtaylor/python-betterproto#459]
If one generates a message with an `optional` field, a group is added around the field.

Input:
```protobuf
syntax = "proto3";

package bug;

message BuggyMessage {
    uint32 field_a = 1;
    optional uint32 field_b = 2;
};
```

Expected:
```python
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# sources: bug.proto
# plugin: python-betterproto
# This file has been @generated

from dataclasses import dataclass

import betterproto


@dataclass(eq=False, repr=False)
class BuggyMessage(betterproto.Message):
    field_a: int = betterproto.uint32_field(1)
    field_b: int = betterproto.uint32_field(2, optional=True)
```

Actual:
```python
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# sources: bug.proto
# plugin: python-betterproto
# This file has been @generated

from dataclasses import dataclass
from typing import Optional

import betterproto


@dataclass(eq=False, repr=False)
class BuggyMessage(betterproto.Message):
    field_a: int = betterproto.uint32_field(1)
    field_b: Optional[int] = betterproto.uint32_field(
        2, optional=True, group="_field_b"
    )
```

I am not sure why it is done that way, but having the field in a group interferes with the custom flag `--custom_opt=pydantic_dataclasses`: every `None` value will fail because no member of the group will be set.

It seems it's not been a problem for the users of this library, so I'm assuming it is the intended behavior. However, it must have changed recently, because between the merge of PR #406 and it's opening, this was not how it was generated.

**Comment by Gobot1234** [comment]
Nothing touching code generation has been merged to the master branch, so there must have been a bug in the original implementation

**Comment by SamuelYvon** [comment]
Are you sure? We tested our usage with my fork yesterday with success, but today our deployment with upstream failed due to this error :(

I can fix the pydantic related issues by tweaking the group check (ie, if the fields are optional, allow them to be left None) but if you think this is a bug in the original code generation, I can take a look and fix that instead

**Comment by Gobot1234** [comment]
The only code that's been merged semi recently was dropping 3.6 and fixing CI due to that merge https://github.com/danielgtaylor/python-betterproto/commits/master

**Comment by SamuelYvon** [comment]
Is it possible it was purposefully done that way? This would allow a user to check if the field is set using the protobuf's "which" (or whatever it's called) without having to bother with a value (I assume serialization on the wire could give it a value like 0)

**Comment by Gobot1234** [comment]
I was under the impression Message.is_set would fulfil this purpose so this does seem like a bug.

**Comment by SamuelYvon** [comment]
Ok. Looking into it...

**Comment by SamuelYvon** [comment]
Turns out it's intentional:

From `src/betterproto/lib/google/protobuf/__init__.py:683`:

```python
    proto3_optional: bool = betterproto.bool_field(17)
    """
    If true, this is a proto3 "optional". When a proto3 field is optional, it
    tracks presence regardless of field type. When proto3_optional is true,
    this field must be belong to a oneof to signal to old proto3 clients that
    presence is tracked for this field. This oneof is known as a "synthetic"
    oneof, and this field must be its sole member (each proto3 optional field
    gets its own synthetic oneof). Synthetic oneofs exist in the descriptor
    only, and do not generate any API. Synthetic oneofs must be ordered after
    all "real" oneofs. For message fields, proto3_optional doesn't create any
    semantic change, since non-repeated message fields always track presence.
    However it still indicates the semantic detail of whether the user wrote
    "optional" or not. This can be useful for round-tripping the .proto file.
    For consistency we give message fields a synthetic oneof also, even though
    it is not required to track presence. This is especially important because
    the parser can't tell if a field is a message or an enum, so it must always
    create a synthetic oneof. Proto2 optional fields do not set this flag,
    because they already indicate optional with `LABEL_OPTIONAL`.
    """

```

**Comment by SamuelYvon** [comment]
I consider this issue resolved by #460 for my purposes; if you feel like there is nothing else to it feel free to close
